flink-core/src/main/java/org/apache/flink/util/TimeUtils.java

```diff
@@ -77,7 +79,14 @@ public static Duration parseDuration(String text) {
         final String unitLabel = trimmed.substring(pos).trim().toLowerCase(Locale.US);
 
         if (number.isEmpty()) {
-            throw new NumberFormatException("text does not start with a number");
+            try {
+                // Fall back to parse ISO-8601 duration format
+                return Duration.parse(trimmed);
```

Comment:
It seems that the duration could be negative right? But before this it shouldn't be. Is it a problem?

Reply:
How about also throwning an exception if it's a negative duration ?

Reply:
I'd be fine with this, since `testParseDurationInvalid` has banned the negative number